controllers/mirrorpeer_controller.go

```diff
@@ -186,6 +190,8 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		statusErr := r.Client.Status().Update(ctx, &mirrorPeer)
 		if statusErr != nil {
 			logger.Error(statusErr, "Error occured while updating the status of mirrorpeer", "MirrorPeer", mirrorPeer)
+			// Requeue, but don't throw
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
If token is not exchanged we need to requeue. I guess Line 199 also needs this change.